### DISCUSSION TITLE
umb-property-info-button should be part of umb-property

### DISCUSSION THREAD

**Discussion by nathanwoulfe** [umbraco/Umbraco-CMS#11144] [Features and ideas]
Stumbled across the `umbPropertyInfoButton` component in the Block List editor:

![image](https://user-images.githubusercontent.com/3248070/134125349-2861d2b1-37a8-4576-9973-8fe0501b0c86.png)

Only place this exists in the codebase is in the block config overlay, but I think it would be super useful to have in the `umbProperty` component as while the control description property is great, it's really only suited to short text. For properties requiring longer descriptions, the info button would be ideal.

This could be wired up via a new prevalue (under the Appearance header), to flip between the existing description or use the info button.

That then leads me to localization. If a property warrants a description (existing way or using the info button), doesn't it follow that it should be localized? If it's complex, all users should be provided a useful description. Could look at using `@` prefixed strings in the property editor description, localizing in `umbProperty`, and keeping everyone happy. It would add some extra burden when creating document types in a multi-lingual site as descriptions would exist in language files rather than the UI, but would still be optional (and hence a non-breaking change).

**Comment by rasmusjp** [comment]
For localizing, can't you still use `#MyPropertyDescription` in the document type property description and then create a dictionary item called `MyPropertyDescription` and it'll use that?

**Reply by nathanwoulfe** [reply]
Will it? I'm not up with localising via the dictionary, I'm all about Lang files in packages :)

If that works, awesome.
